projects/params_vs_compute/hash_ladder/hash_ladder.py

```diff
@@ -0,0 +1,477 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from __future__ import annotations
+import torch
+from typing import Dict, Optional, Tuple
+import torch.nn as nn
+
+from parlai.agents.transformer.modules import (
+    TransformerDecoder,
+    TransformerDecoderLayer,
+    TransformerGeneratorModel,
+)
+
+from parlai.agents.transformer.modules import (
+    create_position_codes,
+    get_n_positions_from_options,
+    LAYER_NORM_EPS,
+)
+
+from parlai.agents.transformer.transformer import TransformerGeneratorAgent
+from parlai.core.opt import Opt
+from parlai.core.params import ParlaiParser
+from parlai.utils.misc import warn_once
+import torch.nn.functional as F
+
+try:
+    from apex.normalization.fused_layer_norm import FusedLayerNorm as LayerNorm
+
+    APEX_LAYER_NORM = True
```

Comment:
We removed Apex, didn't we? If we need to keep this in this file for reproducibility that's fine, but I imagine it will also end up being removed eventually.

Reply:
removed